flycheck.el

```diff
@@ -5640,7 +5640,10 @@ See URL `http://elixir-lang.org/'."
    (warning line-start (file-name) ":" line ": warning:" (message) line-end)
    ;; Warnings from older Elixir versions
    (warning line-start (file-name) ":" line ": " (message) line-end))
-  :modes elixir-mode)
+  :modes elixir-mode
+  :predicate
+  (lambda ()
+    (not (string-match "\.exs$" buffer-file-name))))
```

Comment:
You need to guard against `buffer-file-name` being `nil`, and please don't use a regular expression here.  Instead, use `file-name-extension` to get the extension and check it with `string-equal`.  That's clearer.

Besides, use `string-match-p` to check whether a regular expression matches a specific string.  `string-match` changes the match data, which isn't needed here.